File: music-application/src/main/java/com/music/sale/application/user/dto/output/UserOutput.java
package com.music.sale.application.user.dto.output;

import com.music.sale.domain.user.enums.UserRole;
import com.music.sale.domain.user.enums.UserStatus;
import lombok.Builder;

import java.time.LocalDateTime;

@Builder
public record UserOutput(
        Long id,
        String nickname,
        String profileImageUrl,
        String email,
        UserRole role,
        UserStatus status,
        boolean isVerified,
        LocalDateTime createdAt
) {
}

File: music-application/src/main/kotlin/com/music/sale/application/user/dto/output/UserOutput.kt
package com.music.sale.application.user.dto.output

import com.music.sale.domain.user.enums.UserRole
import com.music.sale.domain.user.enums.UserStatus
import java.time.LocalDateTime

data class UserOutput(
    val id: Long?,
    val nickname: String?,
    val profileImageUrl: String?,
    val email: String?,
    val role: UserRole?,
    val status: UserStatus?,
    val isVerified: Boolean,
    val createdAt: LocalDateTime?,
)
